Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,103 @@
+
+```bash
+$ K8SREPO="github.com/madhanrm/kubernetes"
+$ go get -d $K8SREPO
+$ cd $GOPATH/src/$K8SREPO
+$ KUBE_BUILD_PLATFORM=windows/amd64 make WHAT=cmd/kubectl
+$ cp ~/_output/bin/kubectl.exe ~/kube/
+$ git checkout cniwindows
+$ make clean && KUBE_BUILD_PLATFORM=windows/amd64 make WHAT=cmd/kubelet
+$ cp ~/_output/bin/kubelet.exe ~/kube/
+$ git checkout winkernelproxy 
+$ KUBE_BUILD_PLATFORM=windows/amd64 make WHAT=cmd/kube-proxy
+$ cp ~/_output/bin/kube-proxy.exe ~/kube/
+```
+
+_You may run into problems with various "permission denied" errors when building these binaries. This is a bug in the Kubernetes build process for Windows, which uses binaries that exist, but are size-zero, and thus don't have executable permissions on them. The solution, unfortunately, is to run `chmod +x _output/bin/*` after every time (~4 times) you hit the error._
+
+Done! Now, we also need the actual Linux Kubernetes binaries for v1.8. You can pull these from [the Kubernetes mainline](https://github.com/kubernetes/kubernetes/releases/tag/v1.8.0-rc.1), or build them from source as above, except using `K8SREPO=k8s.io/kubernetes` and the `release-1.8` branch. Copy these to `~/kube/bin` (note the extra `/bin` compared to the above path; we actual need to use these during master configuration, rather than just copying them to the Windows node).
+
+### Install CNI ###
+We also need to install the basic CNI binaries so that networking works. Download them from [here](https://github.com/containernetworking/plugins/releases) and copy them to `/opt/cni/bin/`.
+
+## Prepare the Master ##
+Copy the entire directory from [this repository](https://github.com/Microsoft/SDN/tree/master/Kubernetes/linux), which contains a lot of the setup scripts and Kubernetes files that are essential to this process. Check them all out to `~/kube/` and make the scripts executable. This entire directory will be getting mounted for a lot of the docker containers in future steps, so keep its structure the same as outlined in the guide. 
+
+We'll be creating the service cluster on the virtual subnet 11.0/16. This isn't configurable, so if you want to tweak this, it'll require manual modification.
+
+### Certificates ###
+First, prepare the certificates that will be used for nodes to communicate inthe cluster. Run `generate-certs.sh` (in the `certs/` directory), passing **your** external IP as its (only) parameter -- this is likely whatever the IP address of your `eth0` interface, or whatever:
+
+    ~/kube/certs $ ./generate-certs.sh 10.123.45.67   # example! replace
+
+### Allow Routing ###
+_This step may be optional, depending on whether or not you've configured your intended cluster subnet to be routable already_.
+
+Windows nodes will each grab a /24 from the cluster, so if your cluster CIDR is 192.168.0.0/16, a Windows node will use 192.168.0.1/24. Just run:
+
+    ~/kube $ generate-routes.sh 192.168

```

Comment:
I pulled down these scripts to my master node by cloning https://github.com/Microsoft/SDN and switching to the k8s-guide branch. The generate-certs.sh script worked fine, but generate-routes.sh and start-kubelet.sh both failed to run properly and it seemed to be related to Windows line endings making their way into those files. I recreated those files manually (copy/pasting the raw content into the file on linux) and then they worked...so plz double check the scripts on GitHub just in case there is something fishy :)

Reply:
Same problem with start-kubeproxy.sh script too:

> root@nick-linux3:~/kube# ./start-kubeproxy.sh 192.160
> ./start-kubeproxy.sh: line 6: $'\r': command not found